crates/oxc_linter/src/rules/typescript/triple_slash_reference.rs

```diff
@@ -0,0 +1,296 @@
+    PreferImport,
+}
+
+impl std::ops::Deref for TripleSlashReference {
+    type Target = TripleSlashReferenceConfig;
+
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+declare_oxc_lint!(
+    /// ### What it does
+    /// Disallow certain triple slash directives in favor of ES6-style import declarations.
+    ///
+    /// ### Why is this bad?
+    /// Use of triple-slash reference type directives is generally discouraged in favor of ECMAScript Module imports.
+    ///
+    /// ### Example
+    /// ```javascript
+    /// /// <reference lib="code" />
+    /// globalThis.value;
+    /// ```
+    TripleSlashReference,
+    correctness
+);
+
+impl Rule for TripleSlashReference {
+    fn from_configuration(value: serde_json::Value) -> Self {
+        let options: Option<&serde_json::Value> = value.get(0);
+        Self(Box::new(TripleSlashReferenceConfig {
+            lib: options
+                .and_then(|x| x.get("lib"))
+                .and_then(serde_json::Value::as_str)
+                .map_or_else(LibOption::default, |value| match value {
+                    "always" => LibOption::Always,
+                    "never" => LibOption::Never,
+                    _ => LibOption::default(),
+                }),
+            path: options
+                .and_then(|x| x.get("path"))
+                .and_then(serde_json::Value::as_str)
+                .map_or_else(PathOption::default, |value| match value {
+                    "always" => PathOption::Always,
+                    "never" => PathOption::Never,
+                    _ => PathOption::default(),
+                }),
+            types: options
+                .and_then(|x| x.get("types"))
+                .and_then(serde_json::Value::as_str)
+                .map_or_else(TypesOption::default, |value| match value {
+                    "always" => TypesOption::Always,
+                    "never" => TypesOption::Never,
+                    "prefer-import" => TypesOption::PreferImport,
+                    _ => TypesOption::default(),
+                }),
+        }))
+    }
+    fn run_once(&self, ctx: &LintContext) {
+        let Some(root) = ctx.nodes().iter().next() else { return };
+        let AstKind::Program(program) = root.kind() else { return };
+
+        // We don't need to iterate over all comments since Triple-slash directives are only valid at the top of their containing file.
+        // We are trying to get the first statement start potioin, falling back to the program end if statement does not exist
+        let comments_range_end = program.body.first().map_or(program.span.end, |v| v.span().start);
+        let comments = ctx.semantic().trivias().comments();
+        let mut refs_for_import = HashMap::new();
+
+        for (start, comment) in comments.range(0..comments_range_end) {
+            let raw = &ctx.semantic().source_text()[*start as usize..comment.end() as usize];
+            if let Some(captures) = REFERENCE_REGEX.captures(raw) {
```

Comment:
Can you try to figure it out without regex. Regex it will hurt performance I think

Reply:
I fixed it!
Indeed, it got better
![Screenshot 2024-01-08 at 22 44 35](https://github.com/oxc-project/oxc/assets/34934746/17722423-33f8-4f04-901c-797da007db3b)